qemu/exec.c

```diff
@@ -1087,8 +1087,8 @@ RAMBlock *qemu_ram_alloc_from_ptr(struct uc_struct *uc, ram_addr_t size, void *h
     RAMBlock *new_block;
     ram_addr_t max_size = size;
 
-    size = HOST_PAGE_ALIGN(uc, size);
```

Comment:
The reason for this fix?

Reply:
In Mac M1 this is returning bad addresses. I'm not sure why and I do not understand this code here, because `size` and `max_size` has exactly the same value at this point.

Reply:
Maybe you can find a test to need this code, if not should we remove it?

Reply:
This coepath I believe is very common and I can't come up with why it doesn't work on M1 since I don't have test environment.